src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3481,6 +3526,225 @@ QIcon ViewProviderSketch::mergeColorfulOverlayIcons (const QIcon & orig) const
+    int32_t* vertices = grid->numVertices.startEditing();
+    for (int i = 0; i < nlines; i++)
+        vertices[i] = 2;
+    grid->numVertices.finishEditing();
+
+    // set the grid coordinates
+    vts->vertex.setNum(2 * nlines);
+    SbVec3f* vertex_coords = vts->vertex.startEditing();
+
+    float minX, minY, maxX, maxY, z;
+    camCenterOnSketch.getValue(minX, minY, z);
+    minX -= (gridDimension / 2);
+    minY -= (gridDimension / 2);
+    maxX = minX + gridDimension;
+    maxY = minY + gridDimension;
+
+    // vertical lines
+    int i_offset_x = static_cast<int>(minX / Step);
+    for (int i = 0; i < vlines; i++) {
+        int iStep = (i + i_offset_x);
+        if (((iStep % numberSubdiv == 0) && divLines) || ((iStep % numberSubdiv != 0) && subDivLines)) {
+            vertex_coords[2 * i].setValue(iStep * Step, minY, 0);
+            vertex_coords[2 * i + 1].setValue(iStep * Step, maxY, 0);
+        }
+        else { 
+            /*the number of vertices is defined before. To know the number of vertices ahead it would require
+            to run the loop once before, which would double computation time.
+            If vertices are not filled then there're visual bugs so there are here filled with dummy values.*/
+            vertex_coords[2 * i].setValue(0, 0, 0);
+            vertex_coords[2 * i + 1].setValue(0, 0, 0);
+        }
+    }
+
+    // horizontal lines
+    int i_offset_y = static_cast<int>(minY / Step) - vlines;
+    for (int i = vlines; i < nlines; i++) {
+        int iStep = (i + i_offset_y);
+        if (((iStep % numberSubdiv == 0) && divLines) || ((iStep % numberSubdiv != 0) && subDivLines)) {
+            vertex_coords[2 * i].setValue(minX, iStep * Step, 0);
+            vertex_coords[2 * i + 1].setValue(maxX, iStep * Step, 0);
+        }
+        else {
+            vertex_coords[2 * i].setValue(0, 0, 0);
+            vertex_coords[2 * i + 1].setValue(0, 0, 0);
+        }
+    }
+    vts->vertex.finishEditing();
+
+    parent->addChild(vts);
+    parent->addChild(grid);
+
+    return GridRoot;
+}
+
+void ViewProviderSketch::handleChangedPropertyType(Base::XMLReader& reader,
+    const char* TypeName,
+    App::Property* prop)
+{
+    Base::Type inputType = Base::Type::fromName(TypeName);
+    if (prop->getTypeId().isDerivedFrom(App::PropertyFloat::getClassTypeId()) &&
+        inputType.isDerivedFrom(App::PropertyFloat::getClassTypeId())) {
+        // Do not directly call the property's Restore method in case the implementation
+        // has changed. So, create a temporary PropertyFloat object and assign the value.
+        App::PropertyFloat floatProp;
+        floatProp.Restore(reader);
+        static_cast<App::PropertyFloat*>(prop)->setValue(floatProp.getValue());
+    }
+    else {
+        ViewProviderPart::handleChangedPropertyType(reader, TypeName, prop);
+    }
+}
+
+void ViewProviderSketch::Restore(Base::XMLReader& reader)
+{
+    ViewProviderPart::Restore(reader);
+}

```

Comment:
What's the reason you added this ?

Reply:
vpsketch inherited from `ViewProvider2DObjectGrid`.
I changed that to 
`class SketcherGuiExport ViewProviderSketch : public PartGui::ViewProvider2DObject`
But as vpSketch inherited from ViewProvider2DObjectGrid, it was receiving the overriden functions from it.
So I moved them from ViewProvider2DObjectGrid to VPSketch. As it no longer inherit them.

Reply:
OK, I'm not sure it's needed actually. I have no exact clue what it's for. @wwmayer may you have some clues about this ?

Reply:
I tracked down the code.

This has to do with a change between FreeCAD 0.15 and 0.16, that App::PropertyDistance was substituted by App:PropertyLength in for property GridSize:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=825861

So basically when reading a file created with FreeCAD 0.15, a property with the name "GridSize" exists, but it is not of the right type. Which is handled by the "handleChangedPropertytype".

The code does an omnibus change for any property deriving from ```PropertyFloat```. I do not know exactly why this was not tailored to the specific type (maybe it was code used at several spots for fixing different property changes).